content/blog/cleaning-up-your-test-data-builders.md

```diff
@@ -0,0 +1,112 @@
+---
+title: "Cleaning Up Your Test Data Builders"
+date: 2019-03-16T13:24:49+01:00
+publishdate: 2019-03-16T13:24:49+01:00
+lastmod: 2019-03-17T00:00:00+01:00
+image: "images/blog/cleaning-up-your-test-data-builders.jpg"
+tags: [ "Cleaner Code", "Test Automation" ]
+comments: true
+draft: true
+---
+
+I still come across a lot of automated tests with many lines of code just to create an object. Even when most data is not relevant for the scenario being tested. Making the tests harder to read and maintain.
+
+Using the [Test Data Builder](http://www.natpryce.com/articles/000714.html) pattern can minimize the lines of code in your test to what's relevant. It's really improved the readability of my test automation code.
+
+I've always had the nagging feeling though that it could be better. Even when using a fairly simple builder, the code seems clunky to me most of the time.
+
+Here's my initial take on a builder to create a simple `Person` object. I only specify a name, gender and address but it's still a lot of code for just three properties.
+
+```csharp
```

Comment:
Licht even kort toe wat je hier doet. Beschrijf wat deze code doet.

Reply:
Bedankt voor deze! Heb hier de code van het laatste voorbeeld gebruikt in natuurlijke taal. Deze aan het eind herhaalt om nut van refactoring kracht bij te zetten.